src/App.ts

```diff
@@ -708,66 +776,149 @@ function buildSource(
 
+  let source: AuthorizeSourceData | OrgAuthorizeSourceData;
   // tslint:disable:max-line-length
-  const source: AuthorizeSourceData = {
-    teamId:
-      type === IncomingEventType.Event || type === IncomingEventType.Command
-        ? ((body as (SlackEventMiddlewareArgs | SlackCommandMiddlewareArgs)['body']).team_id as string)
-        : type === IncomingEventType.Action ||
-          type === IncomingEventType.Options ||
-          type === IncomingEventType.ViewAction ||
-          type === IncomingEventType.Shortcut
-        ? ((body as (
-            | SlackActionMiddlewareArgs
-            | SlackOptionsMiddlewareArgs
-            | SlackViewMiddlewareArgs
-            | SlackShortcutMiddlewareArgs
-          )['body']).team.id as string)
-        : assertNever(type),
-    enterpriseId:
-      type === IncomingEventType.Event || type === IncomingEventType.Command
-        ? ((body as (SlackEventMiddlewareArgs | SlackCommandMiddlewareArgs)['body']).enterprise_id as string)
-        : type === IncomingEventType.Action ||
-          type === IncomingEventType.Options ||
-          type === IncomingEventType.ViewAction ||
-          type === IncomingEventType.Shortcut
-        ? ((body as (
-            | SlackActionMiddlewareArgs
-            | SlackOptionsMiddlewareArgs
-            | SlackViewMiddlewareArgs
-            | SlackShortcutMiddlewareArgs
-          )['body']).team.enterprise_id as string)
-        : undefined,
-    userId:
-      type === IncomingEventType.Event
-        ? typeof (body as SlackEventMiddlewareArgs['body']).event.user === 'string'
-          ? ((body as SlackEventMiddlewareArgs['body']).event.user as string)
-          : typeof (body as SlackEventMiddlewareArgs['body']).event.user === 'object'
-          ? ((body as SlackEventMiddlewareArgs['body']).event.user.id as string)
-          : (body as SlackEventMiddlewareArgs['body']).event.channel !== undefined &&
-            (body as SlackEventMiddlewareArgs['body']).event.channel.creator !== undefined
-          ? ((body as SlackEventMiddlewareArgs['body']).event.channel.creator as string)
-          : (body as SlackEventMiddlewareArgs['body']).event.subteam !== undefined &&
-            (body as SlackEventMiddlewareArgs['body']).event.subteam.created_by !== undefined
-          ? ((body as SlackEventMiddlewareArgs['body']).event.subteam.created_by as string)
-          : undefined
-        : type === IncomingEventType.Action ||
-          type === IncomingEventType.Options ||
-          type === IncomingEventType.ViewAction ||
-          type === IncomingEventType.Shortcut
-        ? ((body as (SlackActionMiddlewareArgs | SlackOptionsMiddlewareArgs | SlackViewMiddlewareArgs)['body']).user
-            .id as string)
-        : type === IncomingEventType.Command
-        ? ((body as SlackCommandMiddlewareArgs['body']).user_id as string)
-        : undefined,
-    conversationId: channelId,
-  };
+  if (body.is_enterprise_install) {
```

Comment:
In the future, we can revisit this long if/else part having duplicated lines. However, as long as this works, it's fine for now.

Reply:
Yes I think in bolt@v3 when we switch to two callbacks for `installationStore` and make `teamId` not required for the non org app install, we can remove the if/else.